mitiq/pec/pec.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (C) 2020 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""High-level probabilistic error cancellation tools."""
+
+from typing import Optional, Callable, Union, Tuple
+import numpy as np
+from mitiq._typing import QPROGRAM
+from mitiq.pec.utils import DecoType
+from mitiq.pec.sampling import sample_circuit
+
+
+def execute_with_pec(
+    circuit: QPROGRAM,
+    executor: Callable[[QPROGRAM], float],
+    deco_dict: DecoType,
```

Comment:
Also (this is not really a blocking review comment but a question) -- just as `execute_with_zne` which has a default factory, should we consider defaulting this to something (e.g. the depolarizing decomposition in #371)? Might be unreasonable, but just a thought.

Reply:
It would make sense to consider the depolarizing noise as default. However the problem is that we wouldn't know how to choose the strength of the depolarizing parameter. So, I guess we cannot set a default decomposition here.